### PR TITLE
DAOS-3144 vos: evict obejct from cache when aggregated

### PR DESCRIPTION
**Pull request by Nasf-Fan**
The vos aggregation logic may delete old object, but it did not
handle the vos_object cache, as to others may access such stale
cached vos_object and cause kinds of strange behavior, such as
server thread segmentation fault.

The patch also renames the vos discard test name to resolve the
test cases number conflict.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

File: src/vos/vos_aggregate.c [hunk 1]
```diff
@@ -239,6 +239,27 @@ vos_agg_obj(daos_handle_t ih, vos_iter_entry_t *entry,
...
+			struct daos_llink	*lret;
+			struct vos_object	*obj;
+
+			lkey.olk_cont = vos_hdl2cont(agg_param->ap_coh);
+			lkey.olk_oid = entry->ie_oid;
+
+			rc = daos_lru_ref_hold(occ, &lkey, sizeof(lkey), NULL,
+					       &lret);
+			if (rc == 0) {
+				obj = container_of(lret, struct vos_object,
```
Comment (NiuYawei) [inline src/vos/vos_aggregate.c]: Sorry, I forgot that object can be got by vos_hdl2oiter(). Please use vos_hdl2oiter() instead of these unnecessary changes.

Comment (Nasf-Fan) [reply src/vos/vos_aggregate.c]: Fixed.

**Comment by Nasf-Fan** [comment]
Rebased against the latest master.
